repose-aggregator/core/repose-core-api/src/main/scala/org/openrepose/core/filter/AbstractConfiguredFilter.scala

```diff
@@ -74,38 +75,42 @@ abstract class AbstractConfiguredFilter[T: ClassTag](val configurationService: C
       this,
       classTag[T].runtimeClass.asInstanceOf[Class[T]]
     )
-
     logger.trace("{} initialized.", this.getClass.getSimpleName)
   }
 
   /**
-    * Unsubscribes from the configuration service.
-    */
-  override def destroy(): Unit = {
-    logger.trace("{} destroying ...", this.getClass.getSimpleName)
-    configurationService.unsubscribeFrom(configFile, this)
-    logger.trace("{} destroyed.", this.getClass.getSimpleName)
-  }
-
-  /**
     * Stores the configuration and marks the filter as initialized.
+    *
     * @param configurationObject
     */
   override def configurationUpdated(configurationObject: T): Unit = {
     logger.trace("{} received a configuration update", this.getClass.getSimpleName)
+    val oldConfiguration = configuration
     configuration = configurationObject
-
+    doConfigurationUpdated(oldConfiguration)
     initialized = true
   }
 
   /**
+    * Called after the configuration reference is updated.
+    *
+    * @param oldConfiguration
+    */
+  def doConfigurationUpdated(oldConfiguration: T): Unit = {
+    logger.trace("{} Default doConfigurationUpdated ...", this.getClass.getSimpleName)
+  }
```

Comment:
Adding a comment for history.

This method makes sense to me for one purpose -- to provide seamless extension of the processing performed by `configurationUpdated`. Specifically, this method means that `configurationUpdated` need not be overridden, so you don't encounter the logging issue where `configurationUpdated` is overridden in the subclass of `AbstractConfiguredFilter` then `super.configurationUpdated` is called.

However, I'm not sure I see the need to store a local variable above and pass the `oldConfiguration` to this method. If you replaced the temporary local variable line with the `doConfigurationUpdated(...)` line, the `doConfigurationUpdated` would still have a reference to both the old and the new configurations. In fact, passing the new configuration as an argument makes more sense to me because it is consistent with the configuration service API, and it is the new configuration is the catalyst for the method invocation, so it seems natural that the new configuration would be parameter to `configurationUpdated`.

Actually, I might also make this method do nothing if you really want seamless extension. Why log the occurrence of literally nothing? Leave meaningful logging to the thing doing meaningful work.